inst/tests/tests.Rraw

```diff
@@ -9898,6 +9898,10 @@ unlink(f)
 test(1761.1, fread("1\n2\n3", fill=TRUE), data.table(V1=1:3))
 test(1761.2, fread("1\n2\n3", fill=FALSE), data.table(V1=1:3))
 
+# CJ should retain timezone information, #2029
+df <- CJ(week=as.POSIXct('2016-01-01', tz = 'UTC'), id=1:10)
+test(1762, attr(df$week, 'tzone'), 'UTC')
```

Comment:
You mean something like`CJ(dt1=as.POSIXct('2016-01-01', tz = 'UTC'), id=1:10, dt2=as.POSIXct(c('2016-01-01', '2016-01-02'), tz = 'UTC'))` (behaves as expected) and various permutations of the order of these columns?